payu/models/cesm_cmeps.py

```diff
@@ -190,6 +185,76 @@ def setup(self):
 
+    def _setup_checks(self):
+        # check pelayout fits within requested cpucount
+        cpucount = int(
+            self.expt.config.get('ncpus', multiprocessing.cpu_count())
+        )
+        all_realms = self.realms
+        for realm in all_realms:
+            ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
+            nthreads = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_nthreads"))
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+
+            if nthreads < 1:
+                raise ValueError(f"The number of {realm}_nthreads ({nthreads}) in "
+                                 f"{NUOPC_CONFIG} must be at least 1.")
+                                 
+            if nthreads > 1:
+                npes = nthreads*ntasks*pestride
+                # this is taken from 
+                # https://github.com/ESCOMP/CMEPS/blob/5b7d76978e2fdc661ec2de4ba9834b985decadc6/cesm/driver/esm.F90#L1007
+                # the correct calculation might be (ntasks-1)*pestride*nthreads + nthreads
+            else:
+                npes = (ntasks-1)*pestride + 1
+
+            if (rootpe + npes) > cpucount:
+                raise ValueError(
+                    f"Insufficient cpus for the {realm} pelayout in {NUOPC_CONFIG}"
+                )
+
+            # check iolayout
+            if realm == "cpl":
+                comp = "MED"  # med and cpl names are both used in runconfig
+            else:
+                comp = realm.upper()
+            if comp in self.runconfig.get_component_list():
+                io_section = f"{comp}_modelio"
+                nc_type = self.runconfig.get(io_section, "pio_typename")
+                ioroot = int(self.runconfig.get(io_section, "pio_root"))
+
+                if ioroot >= npes:
+                    raise ValueError(
+                        f"{io_section} pio_root exceeds available PEs (max: {npes - 1}) "
+                        f"in {NUOPC_CONFIG}."
+                    )
+
+                match nc_type:
+                    case "netcdf":
+                        break
+                    case "netcdf4p" | "pnetcdf":
+                        if self.runconfig.get(io_section, "pio_async_interface") == ".false.":
+                            niotasks = int(self.runconfig.get(io_section, "pio_numiotasks"))
+                            iostride = int(self.runconfig.get(io_section, "pio_stride"))
+                            if (ioroot + (niotasks-1)*iostride) >= npes:
+                                raise ValueError(
+                                    f"The iolayout for {io_section} in {NUOPC_CONFIG} is "
+                                    "requesting out of range cpus"
+                                )
+                        # To-do: add coverage for pio_async == .true.
```

Comment:
If `pio_async == .true.` isn't supported what can we tell the user? Warn them it hasn't been checked for consistency?

Reply:
That doesn't sound like a very useful warning ? ...

"Warning: payu hasn't done something you probably weren't expecting it to do ... ?"

Reply:
I think from a users perspective `payu` is doing a number of consistency checks but for this specific case it doesn't. I think it is useful to tell users this so they don't assume if they have an error this part is fine because `payu` hasn't complained.

Reply:
Ok this has a warning now